pkg/health/queuedProbeWorker.go

```diff
@@ -181,3 +187,27 @@ func checkResponse(response int, expected []int) bool {
 	}
 	return false
 }
+
+// TransportWithDNSResponse creates a new transport which overrides hostnames.
+func TransportWithDNSResponse(overrides map[string]string) http.RoundTripper {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
```

Comment:
I wonder if it makes sense to have these as part of the spec of the health checks. Not necessarily for this PR but might be interesting

Reply:
yeah, that's an interesting idea. Let's track that in a Jira?

Reply:
Sounds good, will do. You mean Jira or a GH issue?